editor/scene/texture/texture_region_editor_plugin.cpp

```diff
@@ -490,26 +492,37 @@ void TextureRegionEditor::_texture_overlay_input(const Ref<InputEvent> &p_input)
 				creating = false;
 			}
 
-		} else if (mb->get_button_index() == MouseButton::RIGHT && mb->is_pressed()) {
-			if (drag) {
-				drag = false;
-				if (edited_margin >= 0) {
-					static Side side[4] = { SIDE_TOP, SIDE_BOTTOM, SIDE_LEFT, SIDE_RIGHT };
-					if (node_ninepatch) {
-						node_ninepatch->set_patch_margin(side[edited_margin], prev_margin);
-					}
-					if (res_stylebox.is_valid()) {
-						res_stylebox->set_texture_margin(side[edited_margin], prev_margin);
-					}
-					edited_margin = -1;
-				} else {
-					_apply_rect(rect_prev);
-					rect = rect_prev;
-					texture_preview->queue_redraw();
-					texture_overlay->queue_redraw();
-					drag_index = -1;
-				}
+		} else if (drag && mb->get_button_index() == MouseButton::RIGHT && mb->is_pressed()) {
+			cancel_drag = true;
+		}
+	}
+
+	Ref<InputEventKey> k = p_input;
+	if (k.is_valid() && k->is_pressed() && k->get_keycode() == Key::ESCAPE) {
```

Comment:
Note that `close_on_escape` uses `ui_cancel`, instead of hard-coding ESCAPE.

Reply:
Changed.